Client/Frontend/Widgets/PhotonActionSheetProtocol.swift

```diff
@@ -224,5 +223,13 @@ extension PhotonActionSheetProtocol {
 
         return [topActions, [copyURL, findInPageAction, toggleDesktopSite, pinToTopSites, sendToDevice, closeTab], [share]]
     }
+
+    func fetchBookmarkStatus(for url: String, completionHandler: @escaping (Bool) -> Void) {
```

Comment:
Or `func fetchBookmarkStatus(for url: String) -> Deferred<Bool>` if we are avoiding completion blocks (I like it as-is because I am a bit old-school, but IIUC seems like we are emphasizing the Deferred way in our code).